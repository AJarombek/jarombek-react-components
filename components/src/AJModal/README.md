### Overview

Reusable modal component that can display custom content.

**Use cases:**

```javascript
import { AJModal } from 'jarombek-react-components';

const CustomComponent = () => 
  <AJModal>
    <div>Modal Content</div>
  </AJModal>
```

### Files

| Filename                   | Description                                                                     |
|----------------------------|---------------------------------------------------------------------------------|
| `AJModal.js`               | The reusable `AJModal` component.                                               |
| `index.js`                 | Index file which exports the `AJModal` component.                               |
| `styles.js`                | JSS styles for the `AJModal` component.                                         |
