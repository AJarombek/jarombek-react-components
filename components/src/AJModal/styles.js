/**
 * Stylesheet for the AJModal component using JSS.
 * Author: Andrew Jarombek
 * Date: 1/5/2020
 */

export default {
  ajModal: {
    background: 'white',
    borderRadius: '3px',
    top: '50%',
    left: '50%',
    position: 'absolute',
    transform: 'translate(-50%, -50%)'
  },
  ajModalBackdrop: {
    width: '100vw',
    height: '100vh',
    zIndex: 1,
    top: 0,
    left: 0,
    overflow: 'hidden',
    position: 'fixed',
    cursor: 'pointer'
  },
  ajModalColorBackdrop: {
    backgroundColor: 'rgba(0, 0, 0, 0.3)'
  }
}
