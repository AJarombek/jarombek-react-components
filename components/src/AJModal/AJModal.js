/**
 * Reusable modal component that can display custom content.
 * @author Andrew Jarombek
 * @since 1/5/2020
 */

import React from 'react';
import PropTypes from 'prop-types';
import {createUseStyles} from 'react-jss';
import classnames from 'classnames';

import styles from './styles';

const useStyles = createUseStyles(styles);

/**
 * Component representing a modal with an optional background.
 * @param children The contents of the modal.
 * @param onClickBackground An action that occurs when the modal is clicked.
 * @param backdrop Whether or not a background is shown behind the modal.
 * @return {*} React elements representing a modal.
 * @constructor
 */
const AJModal = ({children, onClickBackground, backdrop=true}) => {
  const classes = useStyles();

  const eventDefault = { stopPropagation: f=>f };

  let backdropClass = backdrop ?
    classnames(classes.ajModalBackdrop, classes.ajModalColorBackdrop, 'aj-modal-backdrop') :
    classnames(classes.ajModalBackdrop, 'aj-modal-backdrop');

  return (
    <div className={backdropClass} onClick={onClickBackground}>
      <div className={classes.ajModal} onClick={(e=eventDefault) => e.stopPropagation()}>
        {children}
      </div>
    </div>
  );
};

AJModal.propTypes = {
  children: PropTypes.node.isRequired,
  onClickBackground: PropTypes.func,
  backdrop: PropTypes.bool
};

export default AJModal;
