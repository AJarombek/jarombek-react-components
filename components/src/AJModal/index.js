import AJModal from './AJModal';

export default AJModal;
