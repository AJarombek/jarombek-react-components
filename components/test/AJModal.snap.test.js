import React from 'react';
import AJModal from '../src/AJModal/AJModal';
import renderer from 'react-test-renderer';

/**
 * Snapshot test for the {@link ../../components/AJModal} component.
 * @author Andrew Jarombek
 * @since 1/5/2020
 */

it('renders correctly', () => {
  const tree = renderer
    .create(
      <AJModal backdrop={true}>
        <div>Modal Snapshot Test</div>
      </AJModal>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
