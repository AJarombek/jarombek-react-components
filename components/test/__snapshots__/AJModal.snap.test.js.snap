// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ajModalBackdrop-0-2-2 ajModalColorBackdrop-0-2-3 aj-modal-backdrop"
>
  <div
    className="ajModal-0-2-1"
    onClick={[Function]}
  >
    <div>
      Modal Snapshot Test
    </div>
  </div>
</div>
`;
